kunlar/12-kun-Regular-ifodalar/12-kun-regular-ifodalar.md

```diff
@@ -0,0 +1,412 @@
+	 by frankenstein.mail.umich.edu (Cyrus v2.3.8) with LMTPA;
+	 Sat, 05 Jan 2008 09:14:16 -0500
+X-Sieve: CMU Sieve 2.3
+Received: from murder ([unix socket])
+	 by mail.umich.edu (Cyrus v2.2.12) with LMTPA;
+	 Sat, 05 Jan 2008 09:14:16 -0500
+Received: from holes.mr.itd.umich.edu (holes.mr.itd.umich.edu [141.211.14.79])
+	by flawless.mail.umich.edu () with ESMTP id m05EEFR1013674;
+	Sat, 5 Jan 2008 09:14:15 -0500
+Received: FROM paploo.uhi.ac.uk (app1.prod.collab.uhi.ac.uk [194.35.219.184])
+	BY holes.mr.itd.umich.edu ID 477F90B0.2DB2F.12494 ;
+	 5 Jan 2008 09:14:10 -0500
+Received: from paploo.uhi.ac.uk (localhost [127.0.0.1])
+	by paploo.uhi.ac.uk (Postfix) with ESMTP id 5F919BC2F2;
+	Sat,  5 Jan 2008 14:10:05 +0000 (GMT)
+Message-ID: <200801051412.m05ECIaH010327@nakamura.uits.iupui.edu>
+Mime-Version: 1.0
+Content-Transfer-Encoding: 7bit
+Received: from prod.collab.uhi.ac.uk ([194.35.219.182])
+          by paploo.uhi.ac.uk (JAMES SMTP Server 2.1.3) with SMTP ID 899
+          for <source@collab.sakaiproject.org>;
+          Sat, 5 Jan 2008 14:09:50 +0000 (GMT)
+Received: from nakamura.uits.iupui.edu (nakamura.uits.iupui.edu [134.68.220.122])
+	by shmi.uhi.ac.uk (Postfix) with ESMTP id A215243002
+	for <source@collab.sakaiproject.org>; Sat,  5 Jan 2008 14:13:33 +0000 (GMT)
+Received: from nakamura.uits.iupui.edu (localhost [127.0.0.1])
+	by nakamura.uits.iupui.edu (8.12.11.20060308/8.12.11) with ESMTP id m05ECJVp010329
+	for <source@collab.sakaiproject.org>; Sat, 5 Jan 2008 09:12:19 -0500
+Received: (from apache@localhost)
+	by nakamura.uits.iupui.edu (8.12.11.20060308/8.12.11/Submit) id m05ECIaH010327
+	for source@collab.sakaiproject.org; Sat, 5 Jan 2008 09:12:18 -0500
+Date: Sat, 5 Jan 2008 09:12:18 -0500
+X-Authentication-Warning: nakamura.uits.iupui.edu: apache set sender to stephen.marquard@uct.ac.za using -f
+To: source@collab.sakaiproject.org
+From: stephen.marquard@uct.ac.za
+Subject: [sakai] svn commit: r39772 - content/branches/sakai_2-5-x/content-impl/impl/src/java/org/sakaiproject/content/impl
+X-Content-Type-Outer-Envelope: text/plain; charset=UTF-8
+X-Content-Type-Message-Body: text/plain; charset=UTF-8
+Content-Type: text/plain; charset=UTF-8
+X-DSPAM-Result: Innocent
+X-DSPAM-Processed: Sat Jan  5 09:14:16 2008
+X-DSPAM-Confidence: 0.8475
+X-DSPAM-Probability: 0.0000
+</details>
+
+## `^` Karet 
+> `^` caret (ingliz tilida) 
+
+Eng birinchi `^` belgisini o'rganamiz. Bu belgini dehqonchasiga `qator boshidan qidir` deb
+tarjima qilishimiz mumkin. Quyidagi programmada, `regifoda.txt` faylini o'qib, `re.search` funksiyasi 
+orqali har bir qatordan `Jumladan` degan so'zni qidirayapmiz. Agar qidirayotgan so'zimiz, qatorda bo'lsa
+o'sha qatorni ekranga konsolga chiqargin deyapmiz,
+
+```python
+# ^ karet bilan ishlash, qator boshini bildiradi
+import re
+
+fayl = open("regifoda.txt", 'r')
+for qator in fayl.readlines():
+    qator = qator.rstrip()
+    if re.search("^Jumladan", qator):
+        print(qator)
```

Comment:
_Mana shu xatoni oldim ishlamadi._

> Traceback (most recent call last):
>   File "E:\Python\git\python\kunlar\12-kun-Regular-ifodalar\12-kun_tekshirish_n.py", line 6, in <module>
>     for qator in fayl.readlines():
>   File "C:\Users\KORD\AppData\Local\Programs\Python\Python39\lib\encodings\cp1251.py", line 23, in decode
>     return codecs.charmap_decode(input,self.errors,decoding_table)[0]
> UnicodeDecodeError: 'charmap' codec can't decode byte 0x98 in position 55: character maps to <undefined>

_Lekin men bunday qilganimda ishladi._

`fayl = open("regifoda.txt", 'r', encoding="utf8")`

Reply:
Mana shu xatoni oldim ishlamadi.

Traceback (most recent call last):
File "E:\Python\git\python\kunlar\12-kun-Regular-ifodalar\12-kun_tekshirish_n.py", line 6, in
for qator in fayl.readlines():
File "C:\Users\KORD\AppData\Local\Programs\Python\Python39\lib\encodings\cp1251.py", line 23, in decode
return codecs.charmap_decode(input,self.errors,decoding_table)[0]
UnicodeDecodeError: 'charmap' codec can't decode byte 0x98 in position 55: character maps to

Lekin men bunday qilganimda ishladi.
`
fayl = open("regifoda.txt", 'r', encoding="utf8")`

Reply:
qayerda `open` ni qo'llangan bo'lsa hammasi o'zgartirildi. Yani, 
> `fayl = open("regifoda.txt", 'r',encoding='utf8')`